src/cim/components/ClusterDeployment/ClusterDeploymentHostsDiscovery.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import { useFormikContext } from 'formik';
+import { Grid, GridItem, TextContent, Text, Button, ButtonVariant } from '@patternfly/react-core';
+import {
+  DiscoveryInstructions,
+  DiscoveryTroubleshootingModal,
+  FormatDiskWarning,
+  HostsNotShowingLink,
+  VMRebootConfigurationInfo,
+} from '../../../common';
+import { getIsSNOCluster } from '../helpers';
+import MinimalHWRequirements from '../Agent/MinimalHWRequirements';
+import {
+  ClusterDeploymentHostsDiscoveryProps,
+  ClusterDeploymentHostsDiscoveryValues,
+} from './types';
+import { InfraEnvAgentTable } from '../InfraEnv';
+import { AddHostModal, EditBMHModal, EditAgentModal } from '../modals';
+import { AgentK8sResource, BareMetalHostK8sResource } from '../../types';
+
+const ClusterDeploymentHostsDiscovery: React.FC<ClusterDeploymentHostsDiscoveryProps> = ({
+  // clusterDeployment,
+  agentClusterInstall,
+  agents,
+  bareMetalHosts,
+  aiConfigMap,
+  infraEnv,
+  usedHostnames,
+  onValuesChanged,
+  onCreateBMH,
+  onDeleteHost,
+  canDeleteAgent,
+  onSaveAgent,
+  onSaveBMH,
+  onSaveISOParams,
+  onFormSaveError,
+  fetchSecret,
+  fetchNMState,
+  isBMPlatform,
+  getClusterDeploymentLink,
+}) => {
+  const [isDiscoveryHintModalOpen, setDiscoveryHintModalOpen] = React.useState(false);
+  const { values } = useFormikContext<ClusterDeploymentHostsDiscoveryValues>();
+  const [isoModalOpen, setISOModalOpen] = React.useState(false);
+  const [editBMH, setEditBMH] = React.useState<BareMetalHostK8sResource>();
+  const [editAgent, setEditAgent] = React.useState<AgentK8sResource | undefined>();
+
+  const isVM = true; // TODO(mlibra): calculate from agent's inventory
+
+  React.useEffect(() => onValuesChanged?.(values), [values, onValuesChanged]);
+
+  const isSNOCluster = getIsSNOCluster(agentClusterInstall);
+
+  return (
+    <Grid hasGutter>
+      <GridItem>
+        <TextContent>
+          <DiscoveryInstructions isSingleNodeCluster={isSNOCluster} />
+        </TextContent>
+      </GridItem>
+      {!!onCreateBMH && (
```

Comment:
When should this button be available? It is currently missing when editing the cluster to add hosts. IIUC it should be available whenever this is AI flow, right?

Reply:
The whole BMH flow needs to be reviewed. I.e. it seems that BMHs are skipped in the Agent's table in the Cluster wizard at all. Let's do that consistently in a follow-up.